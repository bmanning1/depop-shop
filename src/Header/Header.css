.header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 1rem;
    background-color: red;
}

.depop-logo {
    height: 2rem;
}
