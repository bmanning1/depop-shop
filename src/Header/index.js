import React from 'react';
import PropTypes from 'prop-types';
import Likes from '../Likes';
import logo from '../static/depop_logo.png';
import './Header.css';

const Header = (props) => {
    return (
        <div className="header">
            <img className="depop-logo" src={logo} alt="depop logo" />
            <Likes {...props}/>
        </div>
    )
};

Header.propTypes = {
    likedItems: PropTypes.arrayOf(PropTypes.exact({
        id: PropTypes.number.isRequired,
        title: PropTypes.string.isRequired
    })).isRequired,
    unlikeItem: PropTypes.func.isRequired
};

export default Header;