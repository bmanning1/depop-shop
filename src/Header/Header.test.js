import React from 'react';
import { shallow } from 'enzyme';
import Header from '.';

describe('Header:', () => {
    let component; 
    const likedItems = [
        {
            id: 1,
            title: 'jumper'
        },
        {
            id: 2,
            title: 'trainers'
        },
    ];

    beforeEach(() => {
        component = shallow(
            <Header likedItems={likedItems} unlikeItem={jest.fn()} />
        );
    });

    test('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});