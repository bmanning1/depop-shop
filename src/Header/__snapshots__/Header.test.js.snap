// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: should match snapshot 1`] = `
<div
  className="header"
>
  <img
    alt="depop logo"
    className="depop-logo"
    src="depop_logo.png"
  />
  <Likes
    likedItems={
      Array [
        Object {
          "id": 1,
          "title": "jumper",
        },
        Object {
          "id": 2,
          "title": "trainers",
        },
      ]
    }
    unlikeItem={[MockFunction]}
  />
</div>
`;
